logback-classic/src/main/java/ch/qos/logback/classic/DelegateAppender.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Logback: the reliable, generic, fast and flexible logging framework.
+ * Copyright (C) <year>, QOS.ch. All rights reserved.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation
+ *
+ *   or (per the licensee's choosing)
+ *
+ * under the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation.
+ */
+package ch.qos.logback.classic;
+
+import java.util.Iterator;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.AppenderBase;
+import ch.qos.logback.core.spi.AppenderAttachable;
+import ch.qos.logback.core.spi.AppenderAttachableImpl;
+
+
+/**
+ * Appender which delegates all events to child appenders.
+ *
+ * @author Juan Pablo Santos Rodríguez
+ * @see http://jira.qos.ch/browse/LOGBACK-551
+ */
+public class DelegateAppender extends AppenderBase<ILoggingEvent> implements AppenderAttachable<ILoggingEvent> {
+
+  AppenderAttachableImpl<ILoggingEvent> aai = new AppenderAttachableImpl<ILoggingEvent>();

```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/tony19/logback/file/2372205418/issues/source?bid=3222618&fileBranchId=3222689#l33)